File: Memento/PHP/Editor.php
<?php

require_once('WorkspaceState.php');

class Editor {
    private string $state;

    public function setState($state) {
        $this->state = $state;
    }

    public function getState() {
        return $this->state;
    }

    public function save() {
        return new WorkspaceState($this->getState());
    }

    public function restore(WorkspaceState $workspaceState) {
        $this->state = $workspaceState->getState();
    }

    public function print() {
        echo $this->state . "\n\n";
    }
}

File: Memento/Dart/Editor.dart
import 'WorkspaceState.dart';

class Editor {
  String? _state;

  void setState(String? state) {
    _state = state;
  }

  String? getState() {
    return _state;
  }

  WorkspaceState save() {
    return WorkspaceState(_state);
  }

  void restore(WorkspaceState? state) {
    if (state != null) {
      _state = state.getState;
    }
  }

  void describe() {
    print('${_state}');
  }
}
